lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php

```diff
@@ -229,20 +224,28 @@ public function setTemplateOptions($templateOptions)
     public function getTransport()
     {
         $this->prepareMessage();
-        $mailTransport = $this->mailTransportFactory->create(['message' => clone $this->message]);
+        $mailTransport = $this->mailTransportFactory->create(['message' => clone $this->getMessage()]);
         $this->reset();
 
         return $mailTransport;
     }
 
+    private function getMessage(): MessageInterface
+    {
+        if (!$this->message) {
```

Comment:
Unfortunately this will need to be a new property as to keep backwards compatibility we need to keep the injection of `$this->message` in the `__construct`. In this case creating a new private property and use it here then marke `message` as deprecated.

Reply:
If it will be a new property, old customizations will break, isn't it?

Reply:
With the existing property still being injected via the constructor then this `getMessage` method would never actually call the factory and create the message class and then as such this change would not actually fix the bug. Yes a new property could break old customizations but it should limit the issues caused.